app/controllers/bookings_controller.rb

```diff
@@ -19,7 +19,7 @@ def create
     @booking = Booking.new(booking_params)
     @booking.animal = @animal
     @booking.user = current_user
-    if @booking.save
+    if @booking.start_date < @booking.end_date && @booking.save
```

Comment:
you don't need that first condition as you added that as a validation in your Booking model. it won't be able to save if the end_after_start_date function (see below) is not satisfied. this line thus only needs an 'if @booking.save'

Reply:
Removed that part